### PR TITLE
feat: models provider models disable and usage dropdown & Models page removal

### PR DESCRIPTION
**Pull request by ivyjeong13**
https://github.com/user-attachments/assets/49feeb78-71d5-4657-ac40-ab1423034c81

Screenshot 2024-12-06 at 3 25 36 PM

* removes Model page & unused components
* adds usage dropdown & active switch for a model in the model provider's models dialog
* add z-index for table header so input buttons don't show up over the header while scrolling
* add recommended tag for a model provider

**Note**: Trying to figure out why it's flickering on opening. It's something to do with the UpdateModelUsage SelectContent 🥲

### CODE REVIEW

**Comment by ryanhopperlowe** [comment]
Screenshot 2024-12-06 at 2 50 30 PM

The best kind of PR 🙌

File: ui/admin/app/components/ui/badge.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const badgeVariants = cva(
                 destructive:
                     "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
                 outline: "text-foreground",
+                faded: "border-transparent bg-primary text-primary-foreground shadow text-[10px]",
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/ui/badge.tsx]: nit: can text-size be handled ad-hoc?

```ts
<Badge variant="faded">
    <TypographySmall />
</Badge>
```

File: ui/admin/app/components/model/UpdateModelActive.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { memo, useState } from "react";
+
+import { Model } from "~/lib/model/models";
+import { ModelApiService } from "~/lib/service/api/modelApiService";
+
+import { Switch } from "~/components/ui/switch";
+
+export const UpdateModelActive = memo(function UpdateModelActive({
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/model/UpdateModelActive.tsx]: Purely curious, but why do we need memo here?

**Review by ryanhopperlowe — APPROVED** [review]

File: ui/admin/app/components/model/UpdateModelActive.tsx (hunk 2 above)
Comment (ivyjeong13) [reply ui/admin/app/components/model/UpdateModelActive.tsx]: Feel free to ignore that haha That is me trying to figure out what's causing the rerendering of the table unnecessarily, causing a flicker effect.

File: ui/admin/app/components/model/UpdateModelUsage.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    return (
+        <Select onValueChange={handleModelUsageChange} value={usage}>
+            <SelectTrigger>
+                <SelectValue placeholder="Select Usage..." />
+            </SelectTrigger>
+
+            <SelectContent>
+                {Object.entries(ModelUsage).map(([key, value]) =>
```
Comment (ivyjeong13) [inline ui/admin/app/components/model/UpdateModelUsage.tsx:39]: Looking into why when this input is added, I get the flickering rerendering happening (but doesn't happen at all when this is not there.

**Pull request merged**